tools/pctools/venv_inst.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+# COPY and RUN this script in a container image, to:
+# create a python virtualenv named as $1,
+# pip install the modules provided as parameters from $2 on,
+# collect the resulting dependencies in a resource_$1.txt file
+# copy the txt in a common dir.
+# --- 
+# $1 CLI Cloud type: aws, azure, ec2, openstack
+# $2 resources output file
+# $3,... : install modules
+# ---
+
+echo $0 $*
+
+# parameters check
+[ $# -lt 3 ] && exit 1
+
+# create venv
+CLOUD=$1
+FILE=$2
+
+virtualenv test_$CLOUD; 
+
+cd test_$CLOUD; 
+
+# activate venv
+source bin/activate;
+
+# Versions in venv
+python3 --version
+pip --version
+
+# install the modules
+shift 2 # remove 2 used param
+pip install $@
+
+# install result
+[ $? -ne 0 ] && exit 2
+
+# collect the resulting modules and versions
+pip freeze > $FILE; 
```

Comment:
```suggestion
pip freeze > ../$FILE; 
```

Then you don't need to copy the file over later on, but I leave it to you if you like this or not :-)

Reply:
I'd prefer to create the file in the specific folder dedicated to that cloud type so that in the final container you have the correct status set for each type. 
Indeed, then that copy '..' is only needed to collect all the files in a folder that will be shared at runtime, from where we can get all the files our needs.

Reply:
Up to you, both approaches work just fine :slightly_smiling_face: